src/waypoint.jsx

```diff
@@ -34,11 +34,27 @@ function getCurrentPosition(bounds) {
     return POSITIONS.invisible;
   }
 
+  // top is within the screen
   if (bounds.viewportTop <= bounds.waypointTop &&
       bounds.waypointTop <= bounds.viewportBottom) {
     return POSITIONS.inside;
   }
 
+  // bottom is within the screen
+  if (bounds.viewportTop <= bounds.waypointBottom &&
+      bounds.waypointBottom <= bounds.viewportBottom) {
+    return POSITIONS.inside;
+  }
+
+  // TODO: consider returning more granular values than 'inside',
+  // eg; `partiallyInside`, `fullyInside`, and `fullyEnclosing`
```

Comment:
I don't know your exact use-case, but what you have seems perfectly fine to start with.